source/guides/guides/test-retries.md

```diff
@@ -0,0 +1,210 @@
+
+# How It Works
+
+By default, tests will not retry when they fail. You will need to {% urlHash "enable test retries in your configuration" Configure-Test-Retries %} to use this feature.
+
+When test retries is on, tests will automatically be retried up to X additional times based on your desired configuration. For example, if test retries is set to `2`, it will retry tests up to 2 additional times (for a total of 3 attempts) before potentially being marked as a failed test.
+
+When each test is run again, the following {% url "hooks" writing-and-organizing-tests#Hooks %} will be re-run also:
+
+- `beforeEach`
+- `afterEach`
+
+{% note warning %}
+However, `before` and `after` hooks will not be run in next attempts.
+{% endnote %}
+
+**The following is a detailed step-by-step example of how test retries works:**
+
+Assuming we have set test retries to `2` (for a total of 3 attempts), the following is how the tests would run.
+
+1. A test runs for the first time. If the {% fa fa-check-circle green %} test passes, Cypress will move forward with any remaining tests as usual.
+
+2. If the {% fa fa-times red %} test fails, Cypress will tell you that the first attempt failed and will to run the test a second time.
+
+{% img /img/guides/test-retries/attempt-2-start.png %}
+
+3. If the {% fa fa-check-circle green %} test passes after the second attempt, Cypress will continue with any remaining tests.
+
+4. If the {% fa fa-times red %} test fails a second time, it will make the final third attempt to re-run the test.
+
+{% img /img/guides/test-retries/attempt-3-start.png %}
+
+5. If the {% fa fa-times red %} test fails a third time, Cypress will mark the test as failed and then move on to run any remaining tests.
+
+{% img /img/guides/test-retries/attempt-3-fail.png %}
+
+The following is a screen capture of what test retries looks like on the same failed test when run via {% url "`cypress run`" command-line#cypress-run %}.
+
+{% img /img/guides/test-retries/cli-error-message.png %}
+
+During {% url "`cypress open`" command-line#cypress-open %} you will be able to see the number of attempts made in the {% url "Command Log" test-runner#Command-Log %} and expand each attempt for review and debugging if desired.
+
+{% video local /img/guides/test-retries/attempt-expand-collapse-time-travel.mp4 %}
+
+# Configure Test Retries
+
+## Global Configuration
+
+If you want to change the number of attempts being made for all tests run for both `cypress run` and `cypress open`, you can configure this in your {% url "configuration file" command-line#cypress-open-config-file-lt-config-file-gt %} (`cypress.json` by default) by defining the `retries` property and giving it the desired number of retries.
+
+```jsx
+{
+  "retries": 1
+}
+```
```

Comment:
```suggestion
```jsx
{
  "retries": {
    "runMode": 2,
    "openMode": 0
  }
}
```

this is recommended global configuration since users likely don't want retries enabled in open mode (slows down iteration on fixing regressions and writing tests)

Reply:
I think this is explicitly defined below, but maybe the section for modes should come first.